tests/Repository.py

```diff
@@ -1749,6 +1749,27 @@ def testMergeWithConflict(self):
         self.assertEqual(raisedexp.exception.status, 409)
         self.assertEqual(raisedexp.exception.data, {"message": "Merge conflict"})
 
+    def testMergeUpstreamSuccess(self):
+        # Use fork for being able to update it
+        repo = self.g.get_repo("Felixoid/PyGithub")
+        # First one to sync with upstream
+        result = repo.merge_upstream("main")
+        self.assertEqual(result.message, "Successfully fetched and fast-forwarded from upstream PyGithub:main.")
+        # Second one to check it's already synced
+        result = repo.merge_upstream("main")
+        self.assertEqual(result.message, "This branch is not behind the upstream PyGithub:main.")
+
+    def testMergeUpstreamFailure(self):
+        # Use fork for being able to update it
+        repo = self.g.get_repo("Felixoid/PyGithub")
+        with self.assertRaises(github.GithubException) as raisedexp:
+            repo.merge_upstream("doesNotExist")
+        self.assertEqual(raisedexp.exception.status, 404)
```

Comment:
Please assert the exception message as noted in https://github.com/PyGithub/PyGithub/pull/3185#pullrequestreview-2611644580